### PR TITLE
Action open url button renderer

### PR DESCRIPTION
**Pull request by rj100189**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
actionOpenUrl

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 1]
```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
...
+                showCardIcon->Property("width", Formatter() << fontSize);
+                showCardIcon->Property("fillMode", "Image.PreserveAspectFit");
+                showCardIcon->Property("anchors.verticalCenter", "parent.verticalCenter");
+
+                std::string file_path = __FILE__;
+                std::string dir_path = file_path.substr(0, file_path.rfind("\\"));
+                dir_path.append("\\Images\\arrow-down_12.svg");
+                std::replace(dir_path.begin(), dir_path.end(), '\\', '/');
+                showCardIcon->Property("source", "\"" + std::string("file:/") + dir_path + "\"");
+
```
Comment (rj100189) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1862]: We should find a better solution to call image file. I will work on a separate PR to try get rid of this.

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 2]
```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
...
+                {
+                    contentImage->Property("anchors.horizontalCenter", "parent.horizontalCenter");
+                }
+
+                //TODO: Adding dummy image. This should be replaced!
+                std::string file_path = __FILE__;
+                std::string dir_path = file_path.substr(0, file_path.rfind("\\"));
+                dir_path.append("\\Images\\Cat.png");
+                std::replace(dir_path.begin(), dir_path.end(), '\\', '/');
+                contentImage->Property("source", "\"" + std::string("file:/") + dir_path + "\"");
```
Comment (rj100189) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1800]: This is repetitive. We should find a better solution to call image file. I will work on a separate PR to try get rid of this.

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 3]
```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
...
+            {
+                bgRectangle->Property("border.color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : '#007EA8'");
+                bgRectangle->Property("color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : " << buttonId << ".hovered ? '#007EA8' : 'white'");
+                contentText->Property("color", Formatter() << buttonId << ".hovered ? '#FFFFFF' : '#007EA8'");
+            }
+
+            textLayout->AddChild(contentText);
+            buttonElement->Property("background", bgRectangle->ToString());
+
+            if (isShowCardButton)
```
Comment (rj100189) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1848]: This PR does not include the drop down icon color change based on the theme. Will be taken care when we work on Show card

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 4]
```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
...
+
+            auto contentText = std::make_shared<QmlTag>("Text");
+            if (!action->GetTitle().empty())
+            {
+                contentText->Property("text", "\"" + action->GetTitle() + "\"");
+            }
+            contentText->Property("font.pixelSize", Formatter() << fontSize);
+
+            //TODO: Add border color and style: default/positive/destructive
+            if (!Utils::IsNullOrWhitespace(action->GetStyle()) && !Utils::CaseInsensitiveCompare(action->GetStyle(), "default"))
```
Comment (rj100189) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1817]: The color here used for button style is hard-coded and is being taken from css file. Its a TODO to fix along with css.

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp (hunk 2 above)
Comment (jmenghan) [reply source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1800]: should we convert it into function where we pass in the file name as argument and the function returns the complete path !!

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp (hunk 1 above)
Comment (dgharde) [reply source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1862]: We can try adding the images to qrc to make this simpler

Comment (rj100189) [reply source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1862]: yes... That's the idea. Create a qrc file in the project

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp (hunk 2 above)
Comment (rj100189) [reply source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1800]: The idea is to remove this permanently and use a qrc instead

**Review by dgharde — APPROVED** [review]

**Review by ggovardh — APPROVED** [review]

**Review by mbaradi — APPROVED** [review]

**Review by jmenghan — APPROVED** [review]

**Pull request merged**